scripts/run_all_tests.sh

```
@@ -0,0 +1,167 @@
+./bin/repocontext query --entity-type "function" --include-callees --depth 5 > outputs/perf_high_depth.txt
+./bin/repocontext query --entity-type "function" --include-callers --include-callees --max-tokens 0 > outputs/perf_no_limit.txt
+
+echo "=== Running NEW Functionality Tests ==="
+./bin/repocontext query --search "Query*" --include-callers --include-callees --max-tokens 500 --depth 2 > outputs/new_search_with_options.txt
+./bin/repocontext query --entity-type "function" --include-callers --include-callees --max-tokens 500 --depth 2 > outputs/new_entity_with_options.txt
+
+echo "=== Generating Summary ==="
+echo "=== TEST SUMMARY ===" > outputs/test_summary.txt
+echo "Total files generated: $(ls outputs/*.txt | wc -l)" >> outputs/test_summary.txt
+echo "Error files: $(ls outputs/error_*.txt 2>/dev/null | wc -l)" >> outputs/test_summary.txt
+echo "Success files: $(ls outputs/*.txt | grep -v error_ | wc -l)" >> outputs/test_summary.txt
```

Shellcheck (0.10.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)